vnext/Microsoft.ReactNative/Views/FocusZoneViewManager.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "FocusZoneViewManager.h"
+
+#include "ViewControl.h"
+
+#include "DynamicAutomationProperties.h"
+
+#include <Modules/NativeUIManager.h>
+#include <Utils/AccessibilityUtils.h>
+#include <Utils/Helpers.h>
+#include <Utils/PropertyUtils.h>
+
+#include <INativeUIManager.h>
+#include <IReactInstance.h>
+
+#include <winrt/Windows.System.h>
+#include <winrt/Windows.UI.Core.h>
+
+#include <UI.Xaml.Controls.Primitives.h>
+#include <UI.Xaml.Input.h>
+#include <winrt/Windows.ApplicationModel.Resources.Core.h>
+
+namespace winrt {
+using namespace Windows::Foundation;
+using namespace Windows::UI;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Media;
+} // namespace winrt
+
+namespace Microsoft {
+namespace ReactNative {
+
+void FocusZoneViewShadowNode::createView() {
+  Super::createView();
+
+  auto panel = GetViewPanel();
+  m_panelKeyDownRevoker =
+      panel.PreviewKeyDown(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnKeyDown(sender, e); });
+  m_panelGettingFocusRevoker =
+      panel.GettingFocus(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnGettingFocus(sender, e); });
+  m_panelLosingFocusRevoker =
+      panel.LosingFocus(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnLosingFocus(sender, e); });
+}
+
+void FocusZoneViewShadowNode::updateProperties(winrt::Microsoft::ReactNative::JSValueObject &props) {
+  for (const auto &pair : props) {
+    const std::string &propertyName = pair.first;
+    const auto &propertyValue = pair.second;
+
+    if (propertyName == "focusDirection") {
+      if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
```

Comment:
IMO it's better to expect a string than to no-op if we get passed something else. At least this way you'll get a crash and you can fix the bug, rather than trying to figure out why it didn't work as you expected

Reply:
I've looked at the other view managers and they seem to assign some default on failure so I've assigned false here.